### PR TITLE
pre-commit: Add crate to list of ignored codespell words

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

**Comment by Be-ing** [comment]
We already have `pInOut`... how long will this list grow? I don't think making our own continually growing list of exceptions is a great approach.

**Comment by Holzhaus** [comment]
> We already have `pInOut`... how long will this list grow? I don't think making our own continually growing list of exceptions is a great approach.

Moved to another file. Other alternatives are removing the codespell hook or getting rid of the "crate" term in our codebase.

**Comment by daschuer** [comment]
Does pInOut work in a file?
I have some doubts that it does work now that way but not via -L

**Comment by Holzhaus** [comment]
> Does pInOut work in a file?
> I have some doubts that it does work now that way but not via -L

The problem is that you need to specify the exact case it occurs in the dict, and then all matches (case-insensitive) will be filtered. So it won't warn if you types `inout` instead of `input`.

I worked around this issue. Now this generates the regex on the fly

**Comment by daschuer** [comment]
Unfortunately this is still broken 
I have also added the capital "Crate" to the exclude file, but I still get complains: 

src/library/crate/cratefeature2.cpp:250: crate ==> create
src/library/crate/cratefeature2.cpp:309: Crate ==> Create
src/library/crate/cratefeature2.cpp:315: Crate ==> Create
src/library/crate/cratefeature2.cpp:339: crate ==> create
src/library/crate/cratefeature2.cpp:381: crate ==> create
src/library/crate/cratefeature2.cpp:396: crate ==> create
src/library/crate/cratefeature2.cpp:442: crate ==> create
src/library/crate/cratefeature2.cpp:456: crate ==> create
src/library/crate/cratefeature2.cpp:468: crate ==> create
src/library/crate/cratefeature2.cpp:621: crate ==> create
src/library/crate/cratefeature2.cpp:677: crate ==> create

I think the best we can do is to apply the ignore list to the codespell output in the same way we ignore the untouched lines.

**Review by daschuer — CHANGES_REQUESTED** [review]
Or is there a bug in our code?

**Comment by Holzhaus** [comment]
> Or is there a bug in our code?

There was, but I figured that filtering the matches afterwards is probably easier and more performant anyway. Please retest.

File: .codespellignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Crate
+crate
+pInOut
```
Comment (daschuer) [inline .codespellignore:3]: Since we are now after the regex we need to omit the prefix as well.

**Review by daschuer — CHANGES_REQUESTED** [review]

**Review by daschuer — CHANGES_REQUESTED** [review]
I see now log spam: 

crate  {'crate', 'Crate', 'InOut'}
crate  {'crate', 'Crate', 'InOut'}
Crate  {'crate', 'Crate', 'InOut'}
crate  {'crate', 'Crate', 'InOut'}
crate  {'crate', 'Crate', 'InOut'}
Crate  {'crate', 'Crate', 'InOut'}
crate  {'crate', 'Crate', 'InOut'}
crate  {'crate', 'Crate', 'InOut'}

**Review by daschuer — APPROVED** [review]
Thank you. LGTM 
Hopefully this hook does not keep itching

**Pull request merged**